mmv1/third_party/terraform/transport/config.go.tmpl

```diff
@@ -593,7 +648,20 @@ func (c *Config) logGoogleIdentities() error {
 
 // Get a TokenSource based on the Google Credentials configured.
 // If initialCredentialsOnly is true, don't follow the impersonation settings and return the initial set of creds.
-func (c *Config) getTokenSource(clientScopes []string, initialCredentialsOnly bool) (oauth2.TokenSource, error) {
+func (c *Config) getTokenSource(ctx context.Context, clientScopes []string, initialCredentialsOnly bool) (oauth2.TokenSource, error) {
+
+	if c.ExternalCredentialsHcpTerraform != nil {
+		// Token source is created using external credentials
+		// Impersonation isn't done here, so ignore use of initialCredentialsOnly bool
```

Comment:
> Impersonation isn't done here

When using external credentials we _are_ impersonating a service account via WIF, but this is different to the impersonation already present in the provider where Terraform might use service account A as it's identity, and A is used to impersonate service account B which is used to actually provision things.